src/Api/PersistentImplementation.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE ImpredicativeTypes          #-}
+{-# LANGUAGE OverloadedStrings           #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE TypeOperators               #-}
+
+{-# OPTIONS_GHC -Wall -Werror -fno-warn-orphans #-}
+
+module Api.PersistentImplementation
```

Comment:
Oh, cool, `Persistent/Implementations/STM` is even better, because it's distant from `Api/Persistent.hs`. BTW, what is our convention about singular vs plural in names? Would not `Implementation/` be better (no 's')?

But I'd not expose the implementation in `src/Persistent.hs`, precisely because I'd like to clearly mark (by having to import *.Implementation.*) any code that reaches beyond the abstraction barrier and so depends on one of the actual implementations (even if it's the best one, the chosen one), not on our common API for it.